Libplanet/Blockchain/BlockChain.cs

```diff
@@ -972,6 +972,22 @@ internal IReadOnlyList<ActionEvaluation> EvaluateActions(Block<T> block)
 
         internal BlockChain<T> Fork(HashDigest<SHA256> point)
         {
+            if (!Contains(point))
+            {
+                throw new ArgumentException(
+                    $"The block [{point}] doesn't exist.",
+                    nameof(point));
+            }
+
+            Block<T> pointBlock = this[point];
```

Comment:
Shouldn't this reference done after `_rwlock.EnterReadLock`?

Reply:
Since `BlockChain<T>[HashDigest<SHA256>]` has its own lock, I moved it here.